collector/collector.go

```diff
@@ -62,35 +63,29 @@ func NewCollector(factories component.Factories) *Collector {
 	var r io.Reader = f
 	col := &Collector{
 		factories:      factories,
-		parserProvider: parserprovider.NewInMemory(r),
+		mapProvider: parserprovider.NewInMemoryMapProvider(r),
 	}
 	return col
 }
 
-func (c *Collector) Start() error {
+func (c *Collector) Start(ctx context.Context) error {
 	params := service.CollectorSettings{
 		BuildInfo: component.BuildInfo{
 			Command:  "otelcol",
 			Description: "Lambda Collector",
 			Version:  Version,
 		},
-		ParserProvider: c.parserProvider,
+		ConfigMapProvider: c.mapProvider,
 		Factories:      c.factories,
 	}
 	var err error
 	c.svc, err = service.New(params)
 	if err != nil {
 		return err
 	}
-	cmd := service.NewCommand(c.svc)
-	if args, ok := os.LookupEnv("OPENTELEMETRY_COLLECTOR_ARGS"); ok {
-		cmd.SetArgs(strings.Split(args, " "))
-	}
```

Comment:
@Aneurysm9 do we need any sort of replacement for reading this env var?

Reply:
I don't think so.  In this configuration the collector doesn't have any CLI flags to process.  It takes the location of the config file from the environment via `OPENTELEMETRY_COLLECTOR_CONFIG_FILE` and all other configuration happens there.

Reply:
Ahh I see. I confirmed in our docs for ADOT Lambda we only recommend using the `OPENTELEMETRY_COLLECTOR_CONFIG_FILE` env var for config.